frontend/src/queries/nodes/DataTable/renderColumn.tsx

```diff
@@ -58,6 +59,16 @@ export function renderColumn(
         }
         if (typeof value === 'object') {
             if (Array.isArray(value)) {
+                if (value[0] === '__hogql_chart_type' && value[1] === 'sparkline') {
+                    const object: Record<string, any> = {}
+                    for (let i = 0; i < value.length; i += 2) {
+                        object[value[i]] = value[i + 1]
+                    }
+                    if ('results' in object && Array.isArray(object.results)) {
+                        return <TableCellSparkline dataset={{ data: object.results }} />
```

Comment:
Every time we pass `{ data: object.results }` as `dataset`, we're creating a new object. Thus the `useEffect(..., [dataset])` code in sparkline runs every time the component is rendered. Probably better to pass `results` as a prop itself.